File: app/src/main/java/mega/privacy/android/app/fragments/managerFragments/TransfersBaseFragment.java
package mega.privacy.android.app.fragments.managerFragments;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import mega.privacy.android.app.R;
import mega.privacy.android.app.components.SimpleDividerItemDecoration;
import mega.privacy.android.app.lollipop.ManagerActivityLollipop;
import mega.privacy.android.app.lollipop.adapters.RotatableAdapter;
import mega.privacy.android.app.lollipop.managerSections.RotatableFragment;

import static mega.privacy.android.app.components.transferWidget.TransfersManagement.*;

public class TransfersBaseFragment extends RotatableFragment {

    protected ImageView emptyImage;
    protected TextView emptyText;
    private RelativeLayout getMoreQuotaView;
    protected RecyclerView listView;
    protected LinearLayoutManager mLayoutManager;

    protected ManagerActivityLollipop managerActivity;

    protected View initView(LayoutInflater inflater, ViewGroup container) {
        View v = inflater.inflate(R.layout.fragment_transfers, container, false);

        listView = v.findViewById(R.id.transfers_list_view);
        listView.addItemDecoration(new SimpleDividerItemDecoration(context));
        mLayoutManager = new LinearLayoutManager(context);
        listView.setLayoutManager(mLayoutManager);
        listView.setHasFixedSize(true);
        listView.setItemAnimator(null);
        listView.addOnScrollListener(new RecyclerView.OnScrollListener() {
            @Override
            public void onScrolled(@NonNull RecyclerView recyclerView, int dx, int dy) {
                super.onScrolled(recyclerView, dx, dy);
                checkScroll();
            }
        });

        emptyImage = v.findViewById(R.id.transfers_empty_image);
        emptyText = v.findViewById(R.id.transfers_empty_text);
        getMoreQuotaView = v.findViewById(R.id.get_more_quota_view);
        v.findViewById(R.id.get_more_quota_upgrade_button).setOnClickListener(v1 -> ((ManagerActivityLollipop) context).navigateToUpgradeAccount());

        setGetMoreQuotaViewVisibility();

        managerActivity.supportInvalidateOptionsMenu();

        return v;
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);

        managerActivity = (ManagerActivityLollipop) context;
    }

    public void checkScroll() {
        managerActivity.changeActionBarElevation(listView != null && listView.canScrollVertically(-1));
    }

    /**
     * Shows an empty view if there are not transfers
     * and the list if there are.
     *
     * @param size  the size of the list of transfers
     */
    protected void setEmptyView(int size) {
        if (size == 0) {
            emptyImage.setVisibility(View.VISIBLE);
            emptyText.setVisibility(View.VISIBLE);
            listView.setVisibility(View.GONE);
            checkScroll();
        } else {
            emptyImage.setVisibility(View.GONE);
            emptyText.setVisibility(View.GONE);
            listView.setVisibility(View.VISIBLE);
        }
    }

    /**
     * Sets the visibility of the view "Get more quota".
     */
    public void setGetMoreQuotaViewVisibility() {
        if (getMoreQuotaView != null) {
            getMoreQuotaView.setVisibility(isOnTransferOverQuota() ? View.VISIBLE : View.GONE);
        }
    }

    @Override
    protected RotatableAdapter getAdapter() {
        return null;
    }

    @Override
    public void activateActionMode() {

    }

    @Override
    public void multipleItemClick(int position) {

    }

    @Override
    public void reselectUnHandledSingleItem(int position) {

    }

    @Override
    protected void updateActionModeTitle() {

    }
}

File: app/src/main/java/mega/privacy/android/app/fragments/managerFragments/TransfersBaseFragment.kt
package mega.privacy.android.app.fragments.managerFragments

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.view.isVisible
import androidx.fragment.app.viewModels
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.flowWithLifecycle
import androidx.lifecycle.lifecycleScope
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.flow.launchIn
import kotlinx.coroutines.flow.onEach
import mega.privacy.android.app.R
import mega.privacy.android.app.components.ChatDividerItemDecoration
import mega.privacy.android.app.databinding.FragmentTransfersBinding
import mega.privacy.android.app.main.ManagerActivity
import mega.privacy.android.app.main.adapters.RotatableAdapter
import mega.privacy.android.app.main.managerSections.ActiveTransfersState
import mega.privacy.android.app.main.managerSections.RotatableFragment
import mega.privacy.android.app.main.managerSections.TransfersViewModel
import mega.privacy.android.app.main.managerSections.WrapContentLinearLayoutManager
import mega.privacy.android.app.utils.ColorUtils
import mega.privacy.android.app.utils.Util

/**
 * The base transfer fragment
 */
@AndroidEntryPoint
open class TransfersBaseFragment : RotatableFragment() {

    /**
     * [TransfersViewModel] instance
     */
    protected val viewModel by viewModels<TransfersViewModel>()

    /**
     * LayoutManager
     */
    protected lateinit var mLayoutManager: LinearLayoutManager

    /**
     * [ChatDividerItemDecoration]
     */
    protected lateinit var itemDecoration: ChatDividerItemDecoration

    /**
     * FragmentTransfersBinding
     */
    protected lateinit var binding: FragmentTransfersBinding


    /**
     * Initial the view
     *
     * @param inflater LayoutInflater
     * @param container ViewGroup
     *
     * @return view
     */
    protected open fun initView(inflater: LayoutInflater, container: ViewGroup?): View {
        binding = FragmentTransfersBinding.inflate(inflater, container, false)

        itemDecoration = ChatDividerItemDecoration(requireContext())

        mLayoutManager = WrapContentLinearLayoutManager(requireContext())
        binding.transfersListView.run {
            addItemDecoration(itemDecoration)
            layoutManager = mLayoutManager
            setHasFixedSize(true)
            itemAnimator = null
            addOnScrollListener(object : RecyclerView.OnScrollListener() {
                override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
                    super.onScrolled(recyclerView, dx, dy)
                    updateElevation()
                }
            })
        }

        binding.layoutGetMoreQuotaView.getMoreQuotaUpgradeButton.setOnClickListener {
            (requireActivity() as ManagerActivity).navigateToUpgradeAccount()
        }
        setupFlow()
        setGetMoreQuotaViewVisibility()
        requireActivity().invalidateOptionsMenu()
        return binding.root
    }

    private fun setupFlow() {
        viewModel.activeState.flowWithLifecycle(
            lifecycle = viewLifecycleOwner.lifecycle,
            minActiveState = Lifecycle.State.RESUMED
        ).onEach { transfersState ->
            when (transfersState) {
                is ActiveTransfersState.GetMoreQuotaViewVisibility -> {
                    val getMoreQuotaView = binding.layoutGetMoreQuotaView.getMoreQuotaView
                    getMoreQuotaView.isVisible = transfersState.isVisible
                    if (transfersState.isVisible) {
                        if (Util.isDarkMode(requireContext())) {
                            getMoreQuotaView.setBackgroundColor(ColorUtils.getColorForElevation(
                                requireContext(),
                                DARK_MODE_ELEVATION))
                        } else {
                            getMoreQuotaView.setBackgroundResource(
                                R.drawable.white_layout_with_broder_shadow)
                        }
                    }
                }
                else -> {}
            }
        }.launchIn(viewLifecycleOwner.lifecycleScope)
    }

    /**
     * Update elevation when scrolling if needed.
     */
    open fun updateElevation() = (requireActivity() as ManagerActivity).changeAppBarElevation(
        binding.transfersListView.canScrollVertically(DEFAULT_SCROLL_DIRECTION))

    /**
     * Shows an empty view if there are not transfers
     * and the list if there are.
     *
     * @param size  the size of the list of transfers
     */
    protected open fun setEmptyView(size: Int) {
        with(binding) {
            val isEmpty = size == 0
            transfersEmptyImage.isVisible = isEmpty
            transfersEmptyText.isVisible = isEmpty
            transfersListView.isVisible = isEmpty.not()
            if (isEmpty) updateElevation()
        }
    }

    /**
     * Sets the visibility of the view "Get more quota".
     */
    fun setGetMoreQuotaViewVisibility() {
        viewModel.setGetMoreQuotaViewVisibility()
    }

    override fun getAdapter(): RotatableAdapter? = null

    override fun activateActionMode() {
    }

    override fun multipleItemClick(position: Int) {
    }

    override fun reselectUnHandledSingleItem(position: Int) {
    }

    override fun updateActionModeTitle() {
    }

    companion object {
        /**
         * The default value for scroll direction
         */
        const val DEFAULT_SCROLL_DIRECTION = -1
        private const val DARK_MODE_ELEVATION = 6F
    }
}